### PR TITLE
Upgrade backstage from 1.3.0 to 1.14.2

### PR DESCRIPTION
**Pull request by aljesusg**
Upgrade backstage app to be aligned with upstream.

```bash
unlocking @backstage/app-defaults@^1.0.3 ~> 1.3.1
unlocking @backstage/catalog-model@^1.0.3 ~> 1.3.0
unlocking @backstage/core-app-api@^1.0.3 ~> 1.8.0
unlocking @backstage/core-plugin-api@^1.0.3 ~> 1.5.1
unlocking @backstage/integration-react@^1.1.1 ~> 1.1.13
unlocking @backstage/plugin-catalog@^1.3.0 ~> 1.11.1
unlocking @backstage/plugin-catalog-common@^1.0.3 ~> 1.0.13
unlocking @backstage/plugin-catalog-graph@^0.2.18 ~> 0.2.30
unlocking @backstage/plugin-catalog-react@^1.1.1 ~> 1.6.0
unlocking @backstage/plugin-github-actions@^0.5.6 ~> 0.5.18
unlocking @backstage/plugin-permission-react@^0.4.2 ~> 0.4.12
unlocking @backstage/plugin-scaffolder@^1.3.0 ~> 1.13.1
unlocking @backstage/plugin-techdocs@^1.2.0 ~> 1.6.3
unlocking @backstage/plugin-techdocs-react@^1.0.1 ~> 1.1.6
unlocking @backstage/plugin-techdocs-module-addons-contrib@^1.0.1 ~> 1.0.13
unlocking @backstage/test-utils@^1.1.1 ~> 1.3.1
unlocking @backstage/catalog-client@^1.0.3 ~> 1.4.1
unlocking @backstage/config@^1.0.1 ~> 1.0.7
unlocking @backstage/plugin-app-backend@^0.3.33 ~> 0.3.45
unlocking @backstage/plugin-catalog-backend@^1.2.0 ~> 1.9.1
unlocking @backstage/plugin-proxy-backend@^0.2.27 ~> 0.2.39
unlocking @backstage/plugin-scaffolder-backend@^1.3.0 ~> 1.14.0
unlocking @backstage/plugin-techdocs-backend@^1.1.2 ~> 1.6.2
unlocking @backstage/cli-common@^0.1.9 ~> 0.1.12
unlocking @backstage/config-loader@^1.1.2 ~> 1.3.0
unlocking @backstage/errors@^1.0.0 ~> 1.1.5
unlocking @backstage/integration@^1.2.1 ~> 1.4.5
unlocking @backstage/plugin-auth-node@^0.2.2 ~> 0.2.14
unlocking @backstage/plugin-scaffolder-common@^1.1.1 ~> 1.3.0
unlocking @backstage/plugin-techdocs-node@^1.1.2 ~> 1.7.1
unlocking @backstage/types@^1.0.0 ~> 1.0.2
unlocking @backstage/version-bridge@^1.0.1 ~> 1.0.4
bumping @backstage/cli in app to ^0.22.7
bumping @backstage/core-components in app to ^0.13.1
bumping @backstage/plugin-api-docs in app to ^0.9.4
bumping @backstage/plugin-catalog-import in app to ^0.9.8
bumping @backstage/plugin-org in app to ^0.6.8
bumping @backstage/plugin-search in app to ^1.3.1
bumping @backstage/plugin-search-react in app to ^1.6.1
bumping @backstage/plugin-tech-radar in app to ^0.6.4
bumping @backstage/plugin-user-settings in app to ^0.7.3
bumping @backstage/theme in app to ^0.3.0
bumping @backstage/cli in backend to ^0.22.7
bumping @backstage/backend-common in backend to ^0.18.5
bumping @backstage/backend-tasks in backend to ^0.5.2
bumping @backstage/plugin-auth-backend in backend to ^0.18.3
bumping @backstage/plugin-permission-common in backend to ^0.7.5
bumping @backstage/plugin-permission-node in backend to ^0.7.8
bumping @backstage/plugin-search-backend in backend to ^1.3.1
bumping @backstage/plugin-search-backend-module-pg in backend to ^0.5.6
bumping @backstage/plugin-search-backend-node in backend to ^1.2.1
bumping @backstage/cli in root to ^0.22.7

Upgraded from release 1.3.0 to 1.14.2, please review these template changes:

  https://backstage.github.io/upgrade-helper/?from=1.3.0&to=1.14.2

```

### CODE REVIEW

File: app-config.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ backend:
   # auth:
   #   keys:
   #     - secret: ${BACKEND_SECRET}
-  baseUrl: /
+  baseUrl: http://localhost:7007
```
Comment (EricWittmann) [inline app-config.yaml:15]: This doesn't seem right.

Comment (andreaTP) [reply app-config.yaml:15]: I really don't remember... I think we were overriding this value

Comment (EricWittmann) [reply app-config.yaml:15]: I'm convinced it's right.  :)

Comment (EricWittmann) [reply app-config.yaml:15]: I'm also convinced it's weird.  lol

**Review by EricWittmann — APPROVED** [review]

**Comment by aljesusg** [comment]
Well. I tested this with my image https://quay.io/repository/aljesusg/backstage-apicurio?tab=tags and works.

In fact the deployment that paolo showed me failed fetching the backend and now works with this so I'm going to merge it

https://backstage-rhaf-apicurio-registry.apps.dev-eng-ocp4-mas.dev.3sca.net/catalog

**Pull request merged**